CMakeLists.txt

```diff
@@ -97,32 +135,56 @@ else()
     message(ERROR "libpowerloader used as neither shared or static library: not supported")
 endif()
 
-# Custom build flags
-if(MSVC)
+target_include_directories(
+    libpowerloader
+    PUBLIC $<BUILD_INTERFACE:${POWERLOADER_INCLUDE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+
+if (MSVC)
+    target_compile_definitions(libpowerloader PUBLIC -DNOMINMAX)
     target_compile_options(libpowerloader
-        PUBLIC
+        PRIVATE
             /wd4275 # Disable warning C4275: We know that we use templates instances in our symbol exports.
             /wd4251 # Disable warning C4251: We know that we use templates instances in our symbol exports.
-#         PRIVATE
+            /permissive-
 #             /W4 # Recommended warning level 4
     )
 else()
-    # TODO: enable tons of warnings to help with security and correctness
+    target_compile_options(libpowerloader
+        PUBLIC
```

Comment:
Should'nt it be `PRIVATE` like for msvc case?

Reply:
ah good catch!